# HG changeset patch
# Parent 6211618dbbb85e16ccdfea2c1c68e098ac1bc333

diff --git a/src/main/java/org/yaml/snakeyaml/constructor/Constructor.java b/src/main/java/org/yaml/snakeyaml/constructor/Constructor.java
--- a/src/main/java/org/yaml/snakeyaml/constructor/Constructor.java
+++ b/src/main/java/org/yaml/snakeyaml/constructor/Constructor.java
@@ -15,7 +15,6 @@
  */
 package org.yaml.snakeyaml.constructor;
 
-import java.beans.IntrospectionException;
 import java.math.BigDecimal;
 import java.math.BigInteger;
 import java.util.ArrayList;
@@ -303,8 +302,7 @@
             return object;
         }
 
-        protected Property getProperty(Class<? extends Object> type, String name)
-                throws IntrospectionException {
+        protected Property getProperty(Class<? extends Object> type, String name) {
             return getPropertyUtils().getProperty(type, name);
         }
     }
